editor/components/inserter/index.js

```diff
@@ -8,36 +8,55 @@ class Inserter extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
 		this.toggle = this.toggle.bind( this );
-		this.close = this.close.bind( this );
+		this.update = true;
+
 		this.state = {
-			opened: false
+			toggle: false
 		};
 	}
 
-	toggle() {
-		this.setState( {
-			opened: ! this.state.opened
-		} );
+	shouldComponentUpdate( nextProps, nextState ) {
+		if ( nextState.toggle && ! this.update ) {
+			this.update = true;
+
+			this.setState( {
+				toggle: false
+			} );
+
+			return false;
+		}
+
+		return true;
 	}
 
-	close() {
-		this.setState( {
-			opened: false
+	toggle( e ) {
+		const toggle = this.inserter.getElementsByClassName( 'editor-inserter__toggle' );
+
+		if ( 'undefined' !== typeof e.currentTarget && toggle[ 0 ] === e.currentTarget.activeElement ) {
+			this.update = false;
+		}
+
+		this.setState( ( prevState ) => {
+			return {
+				toggle: ! prevState.toggle
+			};
 		} );
 	}
 
 	render() {
-		const { opened } = this.state;
+		const { toggle } = this.state;
 		const { position } = this.props;
 
 		return (
-			<div className="editor-inserter">
+			<div className="editor-inserter" ref={ ( inserter ) => {
```

Comment:
Added a `ref` to determine if the current button clicked (when menu is open) is the same button which opened the menu.